lightningd/plugin.c

```diff
@@ -482,7 +483,8 @@ static struct io_plan *plugin_read_json(struct io_conn *conn,
 
 		if (err) {
 			plugin_kill(plugin, "%s", err);
-			return io_close(plugin->stdout_conn);
+			/* plugin_kill frees plugin */
+			return io_close(NULL);
```

Comment:
This is interesting, what's the effect of `io_close(NULL)`? Is it just intended as a no-op?

Reply:
Yes, just like the one above it, it's a cheats way of getting an io object to return